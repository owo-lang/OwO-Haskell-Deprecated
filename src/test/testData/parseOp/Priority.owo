module Priority where

  infixl 1 - +
  infixl 2 / *
  val = 1 + 1 * 1 / 1 - 1
