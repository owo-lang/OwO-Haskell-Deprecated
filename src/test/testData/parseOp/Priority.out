File type: CodeFileType
Top module name: .Priority
 implementation of function  "val" (55 5 3) (58 5 6)
  no pragmas
  clause for  "val" (55 5 3) (58 5 6)
  patterns matching parsed as expression
   named reference "val" (55 5 3) (58 5 6)
  function body
   function application
   function being applied
    function application
    function being applied
     named reference "+" (63 5 11) (64 5 12)
    value applied to the function
     constant (61 5 9) (62 5 10)
      IntegerConst 1
   value applied to the function
    function application
    function being applied
     function application
     function being applied
      named reference "-" (75 5 23) (76 5 24)
     value applied to the function
      function application
      function being applied
       function application
       function being applied
        named reference "*" (67 5 15) (68 5 16)
       value applied to the function
        constant (65 5 13) (66 5 14)
         IntegerConst 1
      value applied to the function
       function application
       function being applied
        function application
        function being applied
         named reference "/" (71 5 19) (72 5 20)
        value applied to the function
         constant (69 5 17) (70 5 18)
          IntegerConst 1
       value applied to the function
        constant (73 5 21) (74 5 22)
         IntegerConst 1
    value applied to the function
     constant (77 5 25) (78 5 26)
      IntegerConst 1
  no with abstraction
  no where clause
